website/docs/en/plugins/webpack/externals-plugin.mdx

```diff
@@ -4,11 +4,94 @@ import { ApiMeta } from '@components/ApiMeta.tsx';
 
 <ApiMeta addedVersion={'0.3.3'} />
 
-This plugin can be used to add external dependencies.
+This plugin allows you to specify external dependencies that should not be bundled into the output files. This is particularly useful for libraries that are already available globally or managed by other scripts.
 
 ```js
 new rspack.ExternalsPlugin(type, externals);
 ```
 
-- [type](/config/externals#externalstype)
-- [externals](/config/externals#externals-1)
+## Options
+
+### type
+
+**Type:**
+
+```ts
+type ExternalsType =
+  | 'var'
+  | 'module'
+  | 'assign'
+  | 'this'
+  | 'window'
+  | 'self'
+  | 'global'
+  | 'commonjs'
+  | 'commonjs2'
+  | 'commonjs-module'
+  | 'commonjs-static'
+  | 'amd'
+  | 'amd-require'
+  | 'umd'
+  | 'umd2'
+  | 'jsonp'
+  | 'system'
+  | 'promise'
+  | 'import'
+  | 'script'
+  | 'node-commonjs';
+```
+
+Specifies the default type for the `externals`.
+
+For more details, refer to [externalsType](/config/externals#externalstype).
+
+### externals
+
+**Type:**
+
+```ts
+type Externals = ExternalItem[] | ExternalItem;
+
+type ExternalItem =
+  | RegExp
+  | string
+  | (
+      | ((
+          data: ExternalItemFunctionData,
+          callback: (err?: Error | null, result?: ExternalItemValue) => void,
+        ) => void)
+      | ((data: ExternalItemFunctionData) => Promise<ExternalItemValue>)
+    );
+
+type ExternalItemValue =
+  | string[]
+  | boolean
+  | string
+  | {
+      [k: string]: any;
+    };
+
+type ExternalItemFunctionData = {
+  context?: string;
+  contextInfo?: ModuleFactoryCreateDataContextInfo;
+  getResolve?: (
+    options?: ResolveOptions,
+  ) =>
+    | ((
+        context: string,
+        request: string,
+        callback: (err?: Error, result?: string) => void,
+      ) => void)
+    | ((context: string, request: string) => Promise<string>);
+  request?: string;
+};
+
+type ModuleFactoryCreateDataContextInfo = {
+  issuer: string;
+  compiler: string;
+};
+```
+
+**Prevent bundling** of certain `import`ed packages and instead retrieve these _external dependencies_ at runtime.
+
+For more details, refer to [externals](/config/externals#externals-1).
```

Comment:
When should users use this plugin instead of the `externals` config, can we explain this in the document.

Reply:
The `externals` configuration uses the `ExternalsPlugin` internally. Therefore, users can always use the `externals` option directly.

```ts
if (options.externals) {
	new ExternalsPlugin(options.externalsType, options.externals).apply(
		compiler
	);
}
```

Reply:
Get, can we add it to document so users can understand this

Reply:
@chenjiahan i complement more info.